app/src/main/java/com/github/se/eduverse/ui/todo/TodoListScreen.kt

```diff
@@ -0,0 +1,164 @@
+      content = { pd ->
+        Column(modifier = Modifier.fillMaxSize().padding(pd).testTag("todoListScreen")) {
+          AddTodoEntry { name ->
+            val newTodo = Todo(todoListViewModel.getNewUid(), name, status = TodoStatus.ACTUAL)
+            todoListViewModel.addNewTodo(newTodo)
+          }
+          Spacer(modifier = Modifier.height(10.dp))
+          Row(
+              modifier = Modifier.padding(8.dp),
+          ) {
+            TextButton(
+                onClick = { showCompleted = false },
+                enabled = showCompleted,
+                colors =
+                    ButtonColors(Color.White, Color(0xFF217384), Color(0xFF217384), Color.White),
+                modifier = Modifier.testTag("currentTasksButton")) {
+                  Text("Current Tasks", modifier = Modifier.padding(8.dp), fontSize = 20.sp)
+                }
+            TextButton(
+                onClick = { showCompleted = true },
+                enabled = !showCompleted,
+                colors =
+                    ButtonColors(Color.White, Color(0xFF217384), Color(0xFF217384), Color.White),
+                modifier = Modifier.testTag("completedTasksButton")) {
+                  Text("Completed Tasks", modifier = Modifier.padding(8.dp), fontSize = 20.sp)
+                }
+          }
+          LazyColumn(modifier = Modifier.padding(16.dp)) {
+            val todosToShow = if (showCompleted) doneTodos.value else actualTodos.value
+            items(todosToShow.size) { i ->
+              val todo = todosToShow[i]
+              TodoItem(
+                  todo,
+                  { todoListViewModel.setTodoActual(it) },
+                  { todoListViewModel.setTodoDone(it) })
+            }
+          }
+        }
+      })
+}
+
+/**
+ * Composable that represents a single todo item
+ *
+ * @param todo the todo to display
+ * @param onUndo code executed when the done button is clicked while the todo is completed
+ * @param onDone code executed when the done button is clicked while the todo is not completed
+ */
+@Composable
+fun TodoItem(todo: Todo, onUndo: (Todo) -> Unit, onDone: (Todo) -> Unit) {
+  val completed = todo.status == TodoStatus.DONE
+  Card(
+      modifier = Modifier.testTag("todoItem_${todo.uid}").fillMaxWidth().padding(vertical = 4.dp),
+      shape = RoundedCornerShape(16.dp)) {
+        Row(verticalAlignment = Alignment.CenterVertically) {
+          IconButton(
+              onClick = {
+                if (completed) {
+                  onUndo(todo)
+                } else {
+                  onDone(todo)
+                }
+              },
+              modifier = Modifier.testTag("todoDoneButton_${todo.uid}").padding(8.dp).size(24.dp),
+              colors =
+                  IconButtonColors(
+                      if (completed) Color.Green else Color.LightGray,
+                      Color.White,
+                      Color.LightGray,
+                      Color.White)) {
```

Comment:
I think the background color should be transparent and not white, so that it looks good in dark mode. This is just a suggestion, if you think it looks better in white no matter the mode leave it this way

Reply:
Yes you're right and it looks better in transparent, I'll change it